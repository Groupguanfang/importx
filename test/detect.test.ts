import { expect, it } from 'vitest'

const cache = [null, true, false] as const
const deps = [true, false] as const
const isTS = [true, false] as const
const isNativeTsImportSupported = [true, false] as const
const isRuntimeSupportsTsx = [true, false] as const

function f(v: boolean | null) {
  return v == null ? '?' : v ? 'T' : 'F'
}

for (const c of cache) {
  for (const d of deps) {
    for (const t of isTS) {
      for (const n of isNativeTsImportSupported) {
        for (const r of isRuntimeSupportsTsx) {
          it(
            `cache=${f(c)} | deps=${f(d)} | isTS=${f(t)} | nativeTS=${f(n)} | tsx=${f(r)}`,
            async () => {
              const { detectLoaderLogic } = await import('../src/detect')
              const result = detectLoaderLogic(c, d, t, n, r)
              expect(result).toMatchSnapshot()
            },
          )
        }
      }
    }
  }
}
