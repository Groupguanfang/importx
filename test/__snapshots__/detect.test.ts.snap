// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tsx=✅ | cache=✅ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=✅ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❌ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"tsx"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=✅ | cache=❓ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `null`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `null`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=✅ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `null`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `null`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❌ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=module | isTS=✅ | nativeTS=❌ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"bundle-require"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=✅ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=commonjs | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=commonjs | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=module | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=module | isTS=✅ | nativeTS=❌ 1`] = `null`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=module | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=module | isTS=❌ | nativeTS=❌ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=❓ | isTS=✅ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=❓ | isTS=✅ | nativeTS=❌ 1`] = `"jiti"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=❓ | isTS=❌ | nativeTS=✅ 1`] = `"native"`;

exports[`tsx=❌ | cache=❓ | deps=❌ | type=❓ | isTS=❌ | nativeTS=❌ 1`] = `"native"`;
